android/objectPermission/app/src/main/java/io/realm/todo/PermissionHelper.java

```diff
@@ -2,33 +2,21 @@
 
 import io.realm.Realm;
 import io.realm.RealmResults;
-import io.realm.SyncUser;
 import io.realm.sync.permissions.ClassPermissions;
 import io.realm.sync.permissions.Permission;
 import io.realm.sync.permissions.RealmPermissions;
-import io.realm.sync.permissions.Role;
 
 public class PermissionHelper {
 
-    /**
-     * Called after a given user is logged in.
-     * This will ensure the permissions system is setup before starting using the app.
-     *
-     * @param user               current logged in SyncUser.
-     * @param postInitialization block to run after the Role has been added or found, usually a
-     *                           navigation to the next screen.
-     */
-    public static void initializePermissions(SyncUser user, Runnable postInitialization) {
-        initializeRealmPermissions();
-        ensureRoleExists(user, postInitialization);
-    }
-
     /**
      * Configure the permissions on the Realm and each model class.
      * This will only succeed the first time that this code is executed. Subsequent attempts
      * will silently fail due to `canSetPermissions` having already been removed.
+     *
+     * @param postInitialization block to run after the Role has been added or found, usually a
+     *                           navigation to the next screen.
      */
-    private static void initializeRealmPermissions() {
+    public static void initializePermissions(Runnable postInitialization) {
```

Comment:
I think it's also necessary to remove update permissions from the `__Role` table to prevent a malicious user from adding themselves to another user's private role.